### PR TITLE
MOB-1417 - Contact requests

### PR DESCRIPTION
**Pull request by Oleg-Pecheneg**
Actual approve request doesn't work in this PR. 
The issue is on the SDK side and will be fixed alongside SDK update in future update

### CODE REVIEW

File: unstoppable-ios-app/domains-manager-ios/Modules/Messaging/Chat/ChatViewPresenter.swift [hunk 1]
```diff
@@ -122,48 +154,57 @@ private extension ChatViewPresenter {
...
-                            self.view?.scrollToItem(item, animated: false)
-                        }
-                    })
-                    checkIfUpToDate()
-                }
-                DispatchQueue.main.async {
-                    self.view?.setLoading(active: false)
+                    addMessages(messagesBefore)
+                    
+                    if !messages.first!.isRead,
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Modules/Messaging/Chat/ChatViewPresenter.swift:166]: Is force unwrap here safe?

File: unstoppable-ios-app/domains-manager-ios/Modules/Messaging/ChatsList/Cells/ChatListCell/ChatListCell.swift [hunk 2]
```diff
@@ -80,10 +95,16 @@ private extension ChatListCell {
...
+    func chatNameFrom(userInfo: MessagingChatUserDisplayInfo) -> String {
+        userInfo.displayName
+    }
+    
     func lastMessageTextFrom(message: MessagingChatMessageDisplayInfo) -> String  {
         switch message.type {
         case .text(let description):
             return description.text
+        case .imageBase64:
+            return "Image" // TODO: - Localize
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Modules/Messaging/ChatsList/Cells/ChatListCell/ChatListCell.swift:107]: Please do

File: unstoppable-ios-app/domains-manager-ios/Modules/Messaging/ChatsList/ChatsListViewController.swift [hunk 3]
```diff
@@ -152,11 +215,23 @@ private extension ChatsListViewController {
...
+            addNavViewIfNil()
             navView?.isHidden = false
             navigationItem.rightBarButtonItem = nil
         case .loading:
+            addNavViewIfNil()
             navView?.isHidden = true
             navigationItem.rightBarButtonItem = nil
+        case .requestsList:
+            navigationItem.titleView = nil
+            title = "Chat Requests"
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Modules/Messaging/ChatsList/ChatsListViewController.swift:227]: Extract/localize

File: unstoppable-ios-app/domains-manager-ios/Modules/Messaging/ChatsList/ChatsRequestsListViewPresenter.swift [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+// MARK: - MessagingServiceListener
+extension ChatsRequestsListViewPresenter: MessagingServiceListener {
+    nonisolated func messagingDataTypeDidUpdated(_ messagingDataType: MessagingDataType) {
+        Task { @MainActor in
+            switch messagingDataType {
+            case .chats(let chats, let profile):
+                if profile.id == self.profile.id {
+//                   chatsList != chats {
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Modules/Messaging/ChatsList/ChatsRequestsListViewPresenter.swift:58]: dead code?

**Review by rommex — APPROVED** [review]
with minor suggestion

File: unstoppable-ios-app/domains-manager-ios/Modules/Messaging/Chat/ChatViewPresenter.swift (hunk 1 above)
Comment (Oleg-Pecheneg) [reply unstoppable-ios-app/domains-manager-ios/Modules/Messaging/Chat/ChatViewPresenter.swift:166]: Nice catch, wrapped it

**Pull request merged**